python/tank_vendor/shotgun_deploy/import_handler.py

```diff
@@ -0,0 +1,283 @@
+            # to load it later.
+            module_info = imp.find_module(module_name, [path])
+            self._module_info[module_fullname] = module_info
+        except ImportError:
+            # no module found, fall back to regular import
+            return None
+
+        # since this object is also the "loader" return itself
+        return self
+
+    def load_module(self, module_fullname):
+        """Custom loader.
+
+        Called by python if the find_module was successful.
+
+        For further info, see the docs on `load_module` here:
+            https://docs.python.org/2/library/imp.html#imp.load_module
+
+        :param module_fullname: The fullname of the module to import
+
+        :returns: The loaded module object.
+
+        """
+
+        file_obj = None
+        try:
+            # retrieve the found module info
+            (file_obj, filename, desc) = self._module_info[module_fullname]
+
+            # attempt to load the module. if this fails, allow it to raise
+            # the usual `ImportError`
+            module = imp.load_module(module_fullname, file_obj, filename, desc)
+        finally:
+            # as noted in the imp.load_module docs, must close the file handle.
+            if file_obj:
+                file_obj.close()
+
+            # no need to carry around the module info now that we've loaded it.
+            # once the module is in `sys.modules`, the custom importer will
+            # no longer run.
+            del self._module_info[module_fullname]
+
+        # the module needs to know the loader so that reload() works
+        module.__loader__ = self
+
+        # the module has been loaded from the proper core location!
+        return module
+
+    def set_core_path(self, path):
+        """Set the core path to use.
+
+        This method clears out `sys.modules` of all previously imported modules
+        and reimports them using the new core path. This method locks the global
+        interpreter to prevent problems from modifying sys.modules in a
+        multithreaded context.
+
+        :param path: str path to the core to import from.
+
+        :raises: ValueError - if the supplied path does not exist or is not
+            a valid directory.
+        """
+
+        # the paths are the same. No need to do anything.
+        if path == self.core_path:
+            return
+
+        if not os.path.exists(path):
+            raise ValueError(
+                "The supplied core path is not a valid directory: '%s'."
+                % (path,)
+            )
+
+        # TODO: ensure that the directory looks like core?
+
+        # hold on to the old core namespaces
+        old_namespaces = self._namespaces
+
+        # set the core path internally. now that this is set,
+        self._core_path = path
+
+        # get the new namespaces
+        self._namespaces = [d for d in os.listdir(path)
+                            if not d.startswith(".")]
```

Comment:
If there is a foo.py file, that'll confuse this, right?

Is there a way to compare sys.modules before/after to see what changes?  Or somehow let python figure out the real set of modules to worry about for us?